src/Features/Core/Portable/Diagnostics/IDiagnosticService.cs

```diff
@@ -4,13 +4,16 @@
 using System.Collections.Generic;
 using System.Threading;
 using Microsoft.CodeAnalysis.Common;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Diagnostics
 {
     internal interface IDiagnosticService
     {
         /// <summary>
         /// Event to get notified as new diagnostics are discovered by IDiagnosticUpdateSource
+        /// 
+        /// These events are serialized to preserve event ordering using <see cref="EventMap"/> and <see cref="SimpleTaskQueue"/>
```

Comment:
👍 This is good.

In the future, for API definitions keep in mind you can leave out the implementation details while still giving usable documentation/constraints. For example:

> Notifications for this event are serialized to preserve order.

If the callbacks are known to occur on different threads, you can expand on the above statement to avoid misinterpretation:

> Notifications for this event are serialized to preserve order. However, individual event notifications may occur on any thread.